internal/pkg/instrumentation/manager.go

```diff
@@ -161,29 +161,26 @@ func (m *Manager) Run(ctx context.Context, target *process.TargetDetails) error
 	for {
 		select {
 		case <-ctx.Done():
-			m.logger.V(1).Info("shutting down all probes due to context cancellation")
-			err := m.cleanup(target)
-			err = errors.Join(err, ctx.Err())
-			m.closingErrors <- err
-			return nil
-		case <-m.done:
-			m.logger.V(1).Info("shutting down all probes due to signal")
+			m.logger.V(1).Info("Shutting down all probes")
 			err := m.cleanup(target)
-			m.closingErrors <- err
-			return nil
-		case e := <-m.incomingEvents:
+
+			wg.Wait()
+			close(eventCh)
```

Comment:
Maybe worth adding a comment here about the importance of closing this channel only after the probes are closed to avoid panic